tests/integration/defs/stress_test/stress_test.py

```diff
@@ -0,0 +1,941 @@
+from glob import glob
+from typing import List, Optional, Tuple
+
+import pandas as pd
+import pytest
+import requests
+from defs.conftest import get_device_count, get_device_memory, llm_models_root
+from defs.trt_test_alternative import (Popen, altered_env, cleanup_process_tree,
+                                       print_info, print_warning)
+
+# Define a constant for process termination timeouts
+GRACEFUL_TERMINATION_TIMEOUT = 4  # seconds
+
+
+@dataclass(frozen=True)
+class ServerConfig:
+    """Dataclass to store server configuration for trtllm-serve"""
+    port: int = 8000
+    host: str = "0.0.0.0"
+    wait_interval: int = 10  # seconds
+    max_wait_seconds: int = 360  # 6 mins <- Larger model need longer wait time
+    health_check_timeout: float = 8  # seconds <- Make it smaller than wait_interval
+
+    @property
+    def url(self) -> str:
+        """Get the server URL"""
+        return f"http://localhost:{self.port}"
+
+
+@dataclass(frozen=True)
+class ModelConfig:
+    """Dataclass to store model configuration for stress tests"""
+    model_dir: str
+    tp_size: int
+    memory_requirement: int
+    backend: Optional[str] = None
+
+    def __str__(self) -> str:
+        model_name = os.path.basename(self.model_dir)
+        backend_str = f"_{self.backend}" if self.backend else ""
+        return f"{model_name}_tp{self.tp_size}{backend_str}"
+
+    @property
+    def model_name(self) -> str:
+        """Extract model name from model_dir for genai-perf"""
+        return os.path.basename(self.model_dir)
+
+
+@dataclass(frozen=True)
+class StressTestConfig:
+    """Dataclass to store stress test configuration"""
+    model_config: ModelConfig
+    server_config: ServerConfig
+    # Stress test parameters for stress-test mode
+    stress_time: int = 60  # 1 mins
+    stress_timeout: int = 300  #  5 mins <- stress_timeout must be greater than stress_time
+
+    # Customized stress test parameters for stress-stage-alone mode
+    # Change to True to run customized stress test
+    customized_stress_test: bool = True
+    customized_stress_time: int = 180  # 3 mins
+    customized_stress_timeout: int = 300  #  5 mins <- stress_timeout must be greater than stress_time
+    customized_stress_concurrency: int = 512
+    customized_stress_request_rate: int = 20
+
+    @property
+    def request_count_stress_test(self) -> int:
+        """Calculate request count for stress test"""
+        # Cannot set exact stress time in genai-perf test, WR is set the stress_time as customized value to get request count
+        stress_request_count = self.customized_stress_request_rate * self.customized_stress_time
+        return stress_request_count
+
+
+@dataclass(frozen=True)
+class PerformanceParams:
+    """Dataclass to store test parameters for genai-perf"""
+    input_len_mean: int = 64
+    input_len_std: int = 16
+    output_len_mean: int = 128
+    output_len_std: int = 32
+    test_timeout: int = 3600  # 1 hour
```

Comment:
whats the differences of this test_timeout value with the other 2 timeout values in the StressTestConfig config?